src/openssl.c

```diff
@@ -12031,29 +12076,35 @@ static int EVP_KDF__gc(lua_State *L) {
 
 
 static int kdf_derive(lua_State *L) {
-	int nid;
+	LUAOSSL_EVP_KDF_PTR kdf;
 	luaL_Buffer b;
 	EVP_KDF_CTX *kctx, **kctxp;
+#if HAVE_OSSL_PARAM
+	OSSL_PARAM params[15], *param = params;
```

Comment:
Does this need any sort of freeing?

Reply:
Nope! These are all stack-allocated, the later "construct" functions are just filling in the structs.

Reply:
How is that even possible with e.g. `OSSL_PARAM_construct_octet_string`: doesn't that store an arbitrary length string in there for later use?
---
Oh I see it just saves the pointer and length.